jsettlers.logic/src/main/java/jsettlers/ai/construction/BestFisherConstructionPositionFinder.java

```diff
@@ -40,13 +40,13 @@ public BestFisherConstructionPositionFinder(EBuildingType buildingType) {
 	@Override
 	public ShortPoint2D findBestConstructionPosition(AiStatistics aiStatistics, AbstractConstructionMarkableMap constructionMap, byte playerId) {
 		List<ScoredConstructionPosition> scoredConstructionPositions = new ArrayList<ScoredConstructionPosition>();
-		int fishDistance = Integer.MAX_VALUE;
+
 		for (ShortPoint2D point : aiStatistics.getLandForPlayer(playerId)) {
 			if (constructionMap.canConstructAt(point.x, point.y, buildingType, playerId) && !aiStatistics.blocksWorkingAreaOfOtherBuilding(point,
 					playerId, buildingType)) {
-				ShortPoint2D fishPosition = aiStatistics.getNearestResourcePointForPlayer(point, FISH, playerId, fishDistance);
+				ShortPoint2D fishPosition = aiStatistics.getNearestResourcePointForPlayer(point, FISH, playerId, buildingType.getWorkradius());
 				if (fishPosition != null) {
-					fishDistance = point.getOnGridDistTo(fishPosition);
```

Comment:
... here the distance was decreased.
The idea behind that is that if you have found once a fisher which has a close distance (lets say 3) then in the next loop step only fisher will be searched which are even closer which ... (see next comment)